source/common/network/utility.cc

```diff
@@ -101,6 +102,25 @@ uint32_t portFromUrl(const std::string& url, const std::string& scheme,
 
 } // namespace
 
+std::string Utility::urlFromSocketAddress(const envoy::api::v2::core::SocketAddress& address,
+                                          const std::string& cluster_type) {
+  if (address.resolver_name().empty() ||
+      address.resolver_name() == Config::AddressResolverNames::get().IP) {
+    return fmt::format("tcp://{}:{}", address.address(), address.port_value());
+  } else if (address.resolver_name() == Config::AddressResolverNames::get().SRV) {
+    if (absl::EndsWithIgnoreCase(address.named_port(), "srv")) {
+      return fmt::format("tcp://{}:{}", address.address(), address.named_port());
+    } else {
+      throw EnvoyException(
```

Comment:
Can we validate this property early on, during config ingestion, so we don't need to check it here and also in `SrvResolver::resolve`?